### PR TITLE
Optimize whitespace handling

### PR DESCRIPTION
**Pull request by j9t**
## Summary by CodeRabbit

- **Performance**
  - Improved whitespace processing speed by approximately 10% on the reference corpus.
  - Reduced unnecessary processing while preserving existing output, including tabs, non-breaking spaces, and line breaks.

- **Compatibility**
  - Maintained consistent whitespace behavior across text, hidden inputs, and form controls.

- **Quality**
  - Expanded coverage for whitespace trimming and collapsing scenarios, including whitespace-only content and boundary cases.

- **Release**
  - Updated the package to version 8.4.2.

### CODE REVIEW

File: test/html.test.js [hunk 1]
```diff
@@ -4074,6 +4075,72 @@ describe('HTML', () => {
...
+    assert.strictEqual(collapseWhitespace('  a  ', conservative, true, true, false), ' a ');
+    assert.strictEqual(collapseWhitespace('\ta\t', conservative, true, true, false), '\ta\t');
+    assert.strictEqual(collapseWhitespace(' \t a \t ', conservative, true, true, false), ' a ');
+
+    // Preserved line breaks outrank the whitespace around them
+    assert.strictEqual(collapseWhitespace(' \n a \n ', lineBreaks, true, true, false), '\na\n');
+    assert.strictEqual(collapseWhitespace('  a  ', lineBreaks, true, true, false), 'a');
+
+    // Collapsing all runs composes with trimming
+    assert.strictEqual(collapseWhitespace('  a  \n  b  ', plain, true, true, true), 'a b');
```
[the remark this answers is not in the document]
Comment (j9t) [reply test/html.test.js:4139]: I think this was addressed—check.

Is there no setting for you to recheck automatically?
